packages/app-builder-lib/src/targets/snap/snapBuilder.ts

```diff
@@ -0,0 +1,187 @@
+import { getBinFromUrl, getBin } from "../../binDownload"
+import { exec, log, isEmptyOrSpaces, copyDir, exists } from "builder-util"
+import { copyFile, mkdir, unlink, writeFile, rename, chmod, rm, readdir } from "fs-extra"
+import * as path from "path"
+import { SNAP_ASSETS } from "./snapScripts"
+import { getMksquashfs } from "./linuxTools"
+import { checkSnapcraftVersion } from "builder-util/out/snap"
+
+export interface SnapBuilderOptions extends CommandWrapperOptions {
+  appDir: string
+  output: string
+  arch: string
+
+  icon?: string
+  hooksDir?: string
+  excludedAppFiles?: string[]
+
+  compression?: "xz" | "lzo"
```

Comment:
Are there other compression algorithms? Gzip is very common on Linux.

Reply:
IIRC, the `mksquashfs` that comes from electron-builder-binaries doesn't support all compression algs, but this was a 1:1 copy from app-builder-bin. I'll investigate further

brew `mksquashfs`
```
Compressors available and compressor specific options:
        gzip (default)
        lzo
        lz4
        xz
        zstd
        lzma
```

electron-builder-binaries app-image tools `mksquashfs`
```
Compressors available and compressor specific options:
        gzip (default)
        xz
```

Reply:
Confirming that `xz` | `lzo` is what the `SnapOptions` also provide: https://github.com/electron-userland/electron-builder/blob/ea86783728e5f59ac9625cd34dbc3c92a9d6f8fc/packages/app-builder-lib/src/options/SnapOptions.ts#L134-L137